contracts/finance/VestingWallet.sol

```diff
@@ -31,18 +36,19 @@ contract VestingWallet is Context {
 
     uint256 private _released;
     mapping(address => uint256) private _erc20Released;
-    address private immutable _beneficiary;
     uint64 private immutable _start;
     uint64 private immutable _duration;
 
     /**
-     * @dev Set the beneficiary, start timestamp and vesting duration of the vesting wallet.
+     * @dev Sets the sender as the initial owner, the beneficiary as the pending owner, the start timestamp and the
+     * vesting duration of the vesting wallet.
      */
-    constructor(address beneficiaryAddress, uint64 startTimestamp, uint64 durationSeconds) payable {
+    constructor(address beneficiaryAddress, uint64 startTimestamp, uint64 durationSeconds) payable Ownable(msg.sender) {
```

Comment:
I'd like to discuss doing this:

```suggestion
    constructor(address benefactorAddress, address beneficiaryAddress, uint64 startTimestamp, uint64 durationSeconds) payable Ownable(benefactorAddress) {
```

I think it has a few extra implications we need to consider, but I think this is definitely the way to go because a `VestingWallet` is the kind of contract that might be deployed with immutable clones. In such cases the deployer won't be able to claim the tokens back in case they're never claimed.

I guess the best solution to this is to allow setting a benefactor address (perhaps with a different name)

Reply:
you can present it as a "recovery address" in case the beneficiary doesn't claim ownership of the vesting

```suggestion
    constructor(address beneficiaryAddress, address recoveryAddress, uint64 startTimestamp, uint64 durationSeconds) payable Ownable(recoveryAddress) {
```

Reply:
is "claiming owership of the vesting a taxable event"?

Reply:
I don't think it is, because a user claiming the ownership doesn't have access to transfer the tokens yet.